edgepi-dac.js

```diff
@@ -2,61 +2,67 @@ module.exports = function (RED) {
   const rpc = require("@edgepi-cloud/edgepi-rpc");
 
   function DacNode(config) {
-    // Create new node instance with user config
     RED.nodes.createNode(this, config);
     const node = this;
-    const ipc_transport = "ipc:///tmp/edgepi.pipe";
-    const tcp_transport = `tcp://${config.tcpAddress}:${config.tcpPort}`;
-    const transport =
-      config.transport === "Network" ? tcp_transport : ipc_transport;
-    const voltage = parseFloat(config.voltage);
-    const channel = rpc.DACChannel[config.dacChannel];
+    let voltage = config.voltage;
+    let gain = config.gain;
+    let channel = config.channel;
 
-    const dac = new rpc.DacService(transport);
+    initializeNode(config).then((dac) => {
+      node.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input received" });
+        try {
+          gain = msg.gain ?? gain;

```

Comment:
I have mentioned an issue with  the operator ?? in
https://github.com/EdgePi-Cloud/node-red-edgepi-thermocouple/pull/29#pullrequestreview-1798405803
Please take a look.